### ISSUE TITLE
Chopper should not use generated classes directly?

### ISSUE THREAD

**Issue by charafau** [lejard-h/chopper#5]
Hi, would it be possible to change api slightly?

Currently if we write definition interface 

```dart
@ChopperApi("MyService", baseUrl: "/resources")
abstract class MyServiceDefinition {
 //code here
}
```

`MyService` class is generated and developers import this in their code. Problem with this approach is that if we want to refactor class name, then we need to change import in all places we use that class.

I think chopper could use similar approach to `built_value` - generate class, based on currents name but add `_$` prefix. Then we could specify necessary method to create `MyServiceDefinition` class.

example:

```dart
@ChopperApi(baseUrl: "/resources")
abstract class SampleService {

  /**
 * This should be added by developers by our specified convention
 */
  static SampleService create(ChopperConfig config) {
   return _$SampleService(config);
  }

  @Get("/todos/1")
  Future<TodoDto> getTodo();

}
```

By enforcing static method current problem would be resolved. 

I created something like this in my custom example here:

https://github.com/charafau/retrofit.dart

What do you think about this?

**Comment by lejard-h** [comment]
I actually have no cons about that :)

great idea 👍

**Comment by lejard-h** [comment]
available in `1.1.0` version
